### DISCUSSION TITLE
More complex input/output system

### DISCUSSION THREAD

**Discussion by jamop4** [QodotPlugin/Qodot#123] [Ideas]
Hey everyone,

I was wondering if it would be possible to incorporate some of the input/output logic from Hammer in Qodot.

Hammer allows for some complicated I/O relationships by allowing users to customise things like delay (that affects the time until the output is fired) and parameter overrides.

![image](https://github.com/QodotPlugin/Qodot/assets/140369333/5f8317f6-4768-4b78-9483-518198f658f9)

I am aware that Qodot has a great signal/receiver system but currently I am not sure if it is possible to this sort logic with its current implementation.

At the moment, I have some idea of how to implement something like this using GDScript but it would be great if it was possible using Qodot's scripts, potentially by adding this logic to the signal/receiver entities?

**Comment by Nikoraito** [comment]
I would love this

**Comment by RhapsodyInGeek** [comment]
This is the kind of thing you would set up with your own trigger entities rather than something prebuilt in Qodot. Qodot has some examples of what you can do, but they're all fairly outdated and should also only be viewed as examples. I would say this would fall outside Qodot's Core Scope. It might be something worth considering for the examples rework though.

If you'd like to see how you might pull off some more complicated trigger methods like delays and waits and multitargeting, you can check out my example entities on my repository.

[Target Management Functions](https://github.com/RhapsodyInGeek/just-godot-things/blob/main/Qodot%20Entities/GD%20Script/useful%20functions/target_management.gd)

[Trigger Volume Entity](https://github.com/RhapsodyInGeek/just-godot-things/blob/main/Qodot%20Entities/GD%20Script/trigger/trigger.gd)

[Trigger Volume QodotFGDSolidClass Definition](https://github.com/RhapsodyInGeek/just-godot-things/blob/main/Qodot%20Entities/Qodot%20FGD%20Resources/solid/triggers/trigger_volume.tres)

[Trigger Counter Entity](https://github.com/RhapsodyInGeek/just-godot-things/blob/main/Qodot%20Entities/GD%20Script/trigger/trigger_counter.gd)

[Trigger Counter QodotFGDPointClass Definition](https://github.com/RhapsodyInGeek/just-godot-things/blob/main/Qodot%20Entities/Qodot%20FGD%20Resources/point/trigger_counter.tres)
